cuML/src/dbscan/vertexdeg/algo.h

```diff
@@ -55,7 +55,8 @@ void launcher(const ML::cumlHandle_impl& handle, Pack<value_t> data, int startVe
         (value_t val, 							// current value in gemm matrix
 		int global_c_idx) {						// index of output in global memory
         int acc = val <= eps2;
-        int vd_offset = global_c_idx / n;   // bucket offset for the vertex degrees
+        int vd_offset = global_c_idx - (n * (global_c_idx / n));   // bucket offset for the vertex degrees
```

Comment:
I have to plead ignorance here as to why this change falls out of the max_elems change? (no need for changes, just professing my confusion!)

Reply:
The batching in dbscan has been fundamentally broken since the algorithm was originally released. We have known this, but didn’t know exactly where and why. This is one of the changes necessary to fix it.